### PR TITLE
Possible fix for #77

### PR DESCRIPTION
**Pull request by benedekh**
Hello,

I implemented a possible fix for #77. It makes the application id queryable and replaces the template application id in the queries, when the server is not in production. Some tests are implemented also.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ### 3.0.2 (Next)
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: Put this back please?

File: index.js [hunk 2]
```diff
@@ -128,6 +128,10 @@ var appServer = function(config) {
       });
 
       var endpoint = path.posix.join(normalizedRoot, app.name);
+	  self.express.route(endpoint+'/appid').get(function(req, res){
```
Comment (dblock) [inline index.js]: Something wrong with tabs/spaces here at least.

Now do we really want this with every app server?

**Comment by dblock** [comment]
This seems strange. We rely on a request to figure out the application ID?

**Comment by benedekh** [comment]
Yes, because the website uses a separate controller which does not have the app.id registered automatically. The app.id in index.js is not accessible from the views/test.ejs.

Maybe I get something wrong with the controller in test.ejs, but is there a better solution for this problem?

File: examples/apps/hello_world/index.js [hunk 3]
```diff
@@ -1,10 +1,14 @@
 var alexa = require('alexa-app');
+var pkg = require('./package.json');
+var applicationId = (typeof pkg.alexa == 'object' ? pkg.alexa.applicationId : "");
```
Comment (dblock) [inline examples/apps/hello_world/index.js]: Supposedly we know exactly what is in our package.json, doesn't seem like the condition is necessary?

**Comment by dblock** [comment]
Ok this is a lot easier. Is this what we have to do for all applications? Maybe something should be added in the README of alexa-app?

File: examples/apps/hello_world/index.js (hunk 3 above)
Comment (benedekh) [inline examples/apps/hello_world/index.js]: What happens if someone forgets to put the applicationId in the package.json? At least a condition check is necessary for the alexa object if it exists. (I don't know if for an alexa-app it is mandatory/optional to put the applicationId in the package.json either.)

**Comment by benedekh** [comment]
I tested the solution for the sample apps and as long as the applicationId was in the package.json or it was set as an alternative value (instead of an empty string), it worked. I didn't test for production (verify=true) tough, but as I remember in production the GET interface (intent simulator) is not available. Correct me please, if I'm wrong.

File: examples/apps/hello_world/index.js (hunk 3 above)
Comment (dblock) [inline examples/apps/hello_world/index.js]: But this is a specific app, with a specific package.json.

File: README.md [hunk 4]
```diff
@@ -217,6 +217,11 @@ AlexaAppServer.start({
 
 Each app (skill) is available at a url endpoint on the server, and responds to POST requests from the Echo. If you load an app's endpoint in your browser with a GET request, it will display an echo simulator that can be used to debug your application. With it, you can send different request types to your app, load slots with values you specify, etc and see the actual generated JSON output from your application.
 
+### Show application ID
+
+To show the application ID in the session correctly, set the `id` field of your alexa-app for that value.
```
Comment (dblock) [inline README.md]: Add an example please.

**Comment by dblock** [comment]
Merged via https://github.com/alexa-js/alexa-app-server/commit/6924e903e205bf26d795b3a0d7d8021aa6a41b76, thanks!

**Pull request closed without merging**